workloadDriver/src/main/java/com/vmware/weathervane/workloadDriver/common/core/loadPath/FindMaxLoadPath.java

```diff
@@ -564,6 +584,34 @@ private void moveToNextPhase() {
 		}
 	}
 
+	private long niceRound(long number, boolean alwaysRoundDown) {
```

Comment:
Another thing that I realized from running a micro test with this branch (which ended up at 1118 users) is that by only rounding the steps we still won't necessarily end up with rounded results, since the final steps may be small and not get rounded.

Perhaps rather than rounding the step size, we should be rounding the value of curUsers after the curRateStep is applied.  In that case we could always round up towards minFailUsers after a passing interval, and down toward maxPassUsers after a failing interval.  If rounding caused the new value of curUsers to equal minFail (or maxPass depending on the direction), then we would need to skip the rounding (because we would need to run since we aren't within the stopping percentage yet).

Reply:
for rounding 10543 to 11000,
In this range, it is rounding to the nearest 1000, such as  10000,11000.

for rounding 5050 to 5200, 
and for rounding 5455, 5555, 5655 all to 5600,
In this range, it is rounding to the nearest 400, such as  4000,4400,4800,5200,5600,6000.

Before a recent change, it would round to the nearest 500 (in this range), but I impacted that with a related modification.
I will adjust this because 500 is nicer.

The ranges were chosen such that every round (up or down) is less than 5% of the original number.
~5% may seem large, but for this code only rate steps are rounded (not total users), and as such the overall round represents a relatively small adjustment.


I will look into instead rounding the total users instead of the rate step. (at a lesser round than ~5%).  This will require more checks towards the range ends, as mentioned.